packages/schema-generator/API.md

```diff
@@ -12,6 +12,7 @@ export class SchemaGenerator {
 
 // @public (undocumented)
 export type SchemaGeneratorConfig = {
+    secretName: string;

```

Comment:
Why do we require both secretName and connectionUri? Shouldn't those both be optional, so that a caller can specify one or the other?

Reply:
The underlying libraries `schema-generator` package in Gen2 repo and `@ws-amplify/graphql-schema-generator` do not know what secret is or how & where to read it. The reason we input the secret name is the generated file need to populate it in the output file.

Gen2 `@aws-amplify/backend-cli` package makes use of the secret client (`import { SecretClient } from '@aws-amplify/backend-secret';`) to read the values and pass it to the library. We do not want the underlying libraries to know anything about how these secrets are managed.

Reply:
Got it. As long as the customer isn't having to supply both, I am OK with this.